### PR TITLE
feat(formatter): add option to split binary expressions before operators

### LINKED ISSUE #5879: 📎 Break lines before binary operators
### Description

Prettier recently changed its behavior for this https://github.com/prettier/prettier/pull/7111

Overall it's just much easier to read/parse this way.

### Input
```ts
if (a_somewhat_long_var_name || another_long_var_name || a_very_overly_totally_long_var_name || even_more_long_var_names || yet_another_long_var_name) {
```

### Current Behavior
```ts
if (
  a_somewhat_long_var_name ||
  another_long_var_name ||
  a_very_overly_totally_long_var_name ||
  even_more_long_var_names ||
  yet_another_long_var_name
) {
```

### Proposed Behavior
```ts
if (
  a_somewhat_long_var_name
  || another_long_var_name
  || a_very_overly_totally_long_var_name
  || even_more_long_var_names
  || yet_another_long_var_name
) {
```

### PR DESCRIPTION
**Pull request by bavalpey**
## Summary

Fixes #5879

Adds a new option to the javascript formatter, `operator_linebreak` that configures whether linebreaks for binary expressions should occur before or after the binary operator.

Prettier has implemented this: https://github.com/prettier/prettier/pull/7111

## Test Plan

I wrote tests for this that are on par with the other tests for formatting options.
It tests that when the option is enabled, long lines break before the binary operator instead of after.
I based it off of similar formatting options tests, each of which had only one unit test and weren't really testing edge cases. I presume that this is sufficient.

## Additional Information

There are nearly 0 logic changes required for this.

To see how it works: 
For the javascript file

```js
const VERY_LONG_CONDITION_1234123412341234123412341234 = false;

if (VERY_LONG_CONDITION_1234123412341234123412341234 && VERY_LONG_CONDITION_1234123412341234123412341234 && VERY_LONG_CONDITION_1234123412341234123412341234 && VERY_LONG_CONDITION_1234123412341234123412341234) {
  console.log("DONE")
}


```

running with the option ``--javascript-formatter-operator-linebreak=before`` produces

```
  ✖ Formatter would have printed the following content:
  
    1  1 │   const VERY_LONG_CONDITION_1234123412341234123412341234 = false;
    2  2 │   
    3    │ - if·(VERY_LONG_CONDITION_1234123412341234123412341234·&&·VERY_LONG_CONDITION_1234123412341234123412341234·&&·VERY_LONG_CONDITION_1234123412341234123412341234·&&·VERY_LONG_CONDITION_1234123412341234123412341234)·{
    4    │ - ··console.log("DONE")
       3 │ + if·(
       4 │ + → VERY_LONG_CONDITION_1234123412341234123412341234
       5 │ + → &&·VERY_LONG_CONDITION_1234123412341234123412341234
       6 │ + → &&·VERY_LONG_CONDITION_1234123412341234123412341234
       7 │ + → &&·VERY_LONG_CONDITION_1234123412341234123412341234
       8 │ + )·{
       9 │ + → console.log("DONE");
    5 10 │   }
    6    │ - 
    7 11 │   

```

### CODE REVIEW

**Comment by bavalpey** [comment]
I'm not understanding some of the checks that are failing, like the codegen and documentation.

**Comment by bavalpey** [comment]
> > I'm not understanding some of the checks that are failing, like the codegen and documentation.
> > Is there a step I am supposed to do to fix the codegen? (test output: [biomejs/biome/actions/runs/15312085380/job/43092346301?pr=6159](https://github.com/biomejs/biome/actions/runs/15312085380/job/43092346301?pr=6159))
> 
> We have a command that takes care of the codegen: `just gen-all`. For this particular case, you can run `just gen-bindings`. Since you updated the configuration, the schema must be updated.
> 
> Here the section of the contribution guide: https://github.com/biomejs/biome/blob/main/CONTRIBUTING.md#checks

Thanks. I've followed the steps and run this now.

I wasn't sure if I needed to add a changelog for this, as the instructions for changelogs seemed to be directed towards PRs that were adding a new minor release? Surely you would bundle something like this with other changes for a minor release?

Let me know if there's anything additional I need to do to get this in. I'm very keen on having this option in.

**Comment by bavalpey** [comment]
Hmm.
I've no idea why benchmarks are now timing out? All I did was run the command to update the schema.

It worked before I did this: https://github.com/biomejs/biome/actions/runs/15348620153

Now it's failing, saying the action timed out: https://github.com/biomejs/biome/actions/runs/15512128393/job/43686071399?pr=6159

**Review by ematipico — CHANGES_REQUESTED** [review]
Thank you @bavalpey 


The PR looks good. I am going to block it because we're close to ship v2 and we don't want to ship more features than we already have. We will merge it and ship in v1

**Comment by PointSingularity** [comment]
Is this considered for the next release?

**Comment by ematipico** [comment]
> Is this considered for the next release? 

Yes, for the next minor. No ETA

**Comment by PointSingularity** [comment]
Hoping this one comes in the next patch 🤞
(need to migrate the whole codebase from prettier, but don't want to submit a 500 file change PR twice)

**Comment by ematipico** [comment]
Oh, damn, that's on us 😢 apologies! We should have added it to next minor, and forgot about it. Let me create a milestone now so I won't forget.

In the meantime, @bavalpey, can you please point this PR to the `next` branch?

**Comment by bavalpey** [comment]
> In the meantime, @bavalpey, can you please point this PR to the `next` branch?

Done. But there's now conflicts. What's the protocol for resolving these? Does a maintainer generally edit?

**Comment by ematipico** [comment]
If you're in a position to resolve the conflicts, that would be great. Otherwise we do it when we can

**Comment by ematipico** [comment]
Hey @bavalpey , will you be able to resolve the conflicts?

**Comment by bavalpey** [comment]
> Hey @bavalpey , will you be able to resolve the conflicts?

Sure let me take a look. Not sure how involved they are.

**Comment by bavalpey** [comment]
> Hey @bavalpey , will you be able to resolve the conflicts?

Done

**Comment by ematipico** [comment]
Thank you. Can you please add a changeset too? That's missing and we require it

**Comment by bavalpey** [comment]
> Thank you. Can you please add a changeset too? That's missing and we require it

Done. Let me know if it looks sufficient.

**Comment by bavalpey** [comment]
Hmm.. Not sure why the PR I did just to fix the merge conflict would cause an unrelated test to break...
Or a performance regression, for that matter.

**Comment by siketyan** [comment]
@bavalpey You can just update the snapshots to make the failing tests pass. You can ignore the performance regression too

**Comment by bavalpey** [comment]
> @bavalpey You can just update the snapshots to make the failing tests pass. You can ignore the performance regression too

Oh, right. New snapshots would've been added since merge, and I forgot to update them. Thanks!

File: .changeset/mighty-seals-dance.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+---
+"@biomejs/biome": minor
+---
+
+Added a new option to Biome's Javascript formatter, `operator_linebreak`, to configure whether long lines should be broken before or after binary operators.
```
Comment (ematipico) [inline .changeset/mighty-seals-dance.md]:
```suggestion
Added a new option to Biome's JavaScript formatter, `javascript.formatter.operatorLinebreak`, to configure whether long lines should be broken before or after binary operators.
```

File: .changeset/mighty-seals-dance.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+For example:
```
Comment (ematipico) [inline .changeset/mighty-seals-dance.md]:
```suggestion
For example, the following configuration:
```

File: .changeset/mighty-seals-dance.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+{
+  "formatter": {
+    "javascript": {
+      "operatorLinebreak": "before" // defaults to "after"
+    }
+  }
+}
+```
+
+Will cause this JavaScript file
```
Comment (ematipico) [inline .changeset/mighty-seals-dance.md]:
```suggestion
Will cause this JavaScript file:
```

**Review by ematipico — APPROVED** [review]

**Comment by 7Hua** [comment]
`"javascript": {
        "formatter": {
            "indentStyle": "space",
            "indentWidth": 4,
            "bracketSpacing": false,
            "quoteStyle": "single",
            "trailingCommas": "none",
            "arrowParentheses": "asNeeded",
            "operator-linebreak": "before",
        }
    },`。Why am I still unable to use operator-linebreak in biome, and it prompts "Found an unknown key operator-linebreak"?

**Comment by ematipico** [comment]
Because it hasn't been released yet. It will be available in the next minor
